### ISSUE TITLE
No fix notification and mock location under recent Android versions ?

### ISSUE THREAD

**Issue by lumathieu** [eltorio/RtkGps#50]
Hello, I discover the nice RtkGPS app following the doc of [Centipede](https://docs.centipede.fr/docs/Rover_rtklib_android/) to use a Drotek DP0601/D910 assembly, but I encounter 2 problems with the app.
In a first sight the app work very well : the connection to antenna and the Centipede network works properly and gave centimeter accuracy after a little time of initialization.

But the mock location don't seems to work on my Galaxy A51 under Android 11. Until the server is started, when I try to use other apps using location (like OSMAnd or Google Maps), the internal GPS starts and these apps give a location with a classic GPS accuracy (4-20 meters).
RtkGPS+ seems continue to run in background and don't forgot the location if I return on.
I check config, mock location app in developper setup menu is set on RtkGPS, the location is turned on GPS only and the option is checked in the app.

 
In a second time, when I searched if the issue was ever described, I discovered in the **Pull Requests** that a fix-state notification has been added in a previous version. But on my phone I never seen message about fix, despite notifications for the app are allowed.

To compare, I ran tests on other Android and it's the same behaviour under Android 10 (tested on a Galaxy Tab and a Xiaomi Redmi Note 9). And with other GPS apps providing mock location on Android 11, when they run it was displayed an Android System notification about the activation of the functionnality and the providing app.

<img src="https://user-images.githubusercontent.com/66409000/114192688-12fcfa80-994e-11eb-90df-2841bf450fe0.jpg" width="200 px"/>

But no issues under Android 6.0 (Wiko U-Feel Lite), fix-state notification is well displayed and mock location works (but without message) : other apps like OSMAnd give an accurate location without any informations about satellites.

I precise that I have test the original and the custom RtkGPS version from @jancelin repos, but have the same issues.

Have you the same issues ?
(Sorry the screenshots are in French).

**Comment by jancelin** [comment]
Hi,


did you reboot your phone, often this simple step allows the mock location to be taken into account.


best regards!


Julien Ancelin
Administrateur SIG
Unité Expérimentale INRAE
545 route du Bois Maché
17450 Saint Laurent de la Prée
05 46 82 10 50
05 46 82 40 49
Noter ma nouvelle adresse : ***@***.***
________________________________
De : lumathieu ***@***.***>
Envoyé : vendredi 9 avril 2021 16:29
À : eltorio/RtkGps
Cc : Julien Ancelin; Mention
Objet : [eltorio/RtkGps] No fix notification and mock location under recent Android versions ? (#50)


Hello, I discover the nice RtkGPS app following the doc of Centipede<https://docs.centipede.fr/docs/Rover_rtklib_android/> to use a Drotek DP0601/D910 assembly, but I encounter 2 problems with the app.
In a first sight the app work very well : the connection to antenna and the Centipede network works properly and gave centimeter accuracy after a little time of initialization.

But the mock location don't seems to work on my Galaxy A51 under Android 11. Until the server is started, when I try to use other apps using location (like OSMAnd or Google Maps), the internal GPS starts and these apps give a location with a classic GPS accuracy (4-20 meters).
RtkGPS+ seems continue to run in background and don't forgot the location if I return on.
I check config, mock location app in developper setup menu is set on RtkGPS, the location is turned on GPS only and the option is checked in the app.

In a second time, when I searched if the issue was ever described, I discovered in the Pull Requests that a fix-state notification has been added in a previous version. But on my phone I never seen message about fix, despite notifications for the app are allowed.

To compare, I ran tests on other Android and it's the same behaviour under Android 10 (tested on a Galaxy Tab and a Xiaomi Redmi Note 9). And with other GPS apps providing mock location on Android 11, when they run it was displayed an Android System notification about the activation of the functionnality and the providing app.

[https://user-images.githubusercontent.com/66409000/114192688-12fcfa80-994e-11eb-90df-2841bf450fe0.jpg]<https://user-images.githubusercontent.com/66409000/114192688-12fcfa80-994e-11eb-90df-2841bf450fe0.jpg>

But no issues under Android 6.0 (Wiko U-Feel Lite), fix-state notification is well displayed and mock location works (but without message) : other apps like OSMAnd give an accurate location without any informations about satellites.

I precise that I have test the original and the custom RtkGPS version from @jancelin<https://github.com/jancelin> repos, but have the same issues.

Have you the same issues ?
(Sorry the screenshots are in French).

-

**Comment by lumathieu** [comment]
Yes, I have rebooted the phone a first time before made tests and re-tried now, no change unfortunately.

**Comment by harlock974** [comment]
Same issue here with a Samsung Galaxy A50 and Android 11. Other mock location software work as expected (eg Fake GPS).

**Comment by MacGeyer** [comment]
Make mock location work again on recent Android device is a must have !
